scst-sign/configuring.md

```diff
@@ -0,0 +1,244 @@
+>   `spec.verification.exclude.resources.namespaces` section, such as the
+>   `kube-system` namespace.
+>   - If no `ClusterImagePolicy` resource is created all images are admitted into
+>   the cluster with the following warning:
+>     ```
+>     Warning: clusterimagepolicies.signing.run.tanzu.vmware.com "image-policy" not found. Image policy enforcement was not applied.
+>     ```
+>   - For a simpler installation process in a non-production environment
+>   VMware recommends you use the manifest below to create the `ClusterImagePolicy`
+>   resource. This manifest includes a cosign public key which signed the public
+>   cosign v1.2.1 image. The cosign public key validates the specified cosign
+>   images. Container images running in system namespaces are currently not
+>   signed. You must configure the image policy webhook to allow these unsigned
+>   images by adding system namespaces to the
+>   `spec.verification.exclude.resources.namespaces` section.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: signing.run.tanzu.vmware.com/v1alpha1
+kind: ClusterImagePolicy
+metadata:
+  name: image-policy
+spec:
+  verification:
+    exclude:
+      resources:
+        namespaces:
+        - kube-system
+    keys:
+    - name: cosign-key
+      publicKey: |
+        -----BEGIN PUBLIC KEY-----
+        MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEhyQCx0E9wQWSFI9ULGwy3BuRklnt
+        IqozONbbdbqz11hlRJy9c7SG+hdcFl9jE9uE/dwtuwU2MqU9T/cN0YkWww==
+        -----END PUBLIC KEY-----
+    images:
+    - namePattern: gcr.io/projectsigstore/cosign*
+      keys:
+      - name: cosign-key
+EOF
+```
+
+## <a id='proving-credentials-for-the-webhook'></a> Providing credentials for the webhook
+
+There are four ways the webhook reads credentials to authenticate to registries
+protected by authentication, in order:
+1. Reading `imagePullSecrets` directly from the resource being admitted.
+1. Reading `imagePullSecrets` from the service account the resource is running
+as.
+1. Reading a `secretRef` from the `ClusterImagePolicy` resource applied to the
+cluster for the container image name pattern that matches the container being
+admitted.
+1. Reading `imagePullSecrets` from the `image-policy-registry-credentials`
+service account in the `image-policy-system` namespace.
+
+If you use [containerd-configured registry credentials](https://github.com/containerd/containerd/blob/main/docs/cri/registry.md#configure-registry-credentials)
+or another mechanism that causes your resources and service accounts to not
+include an `imagePullSecrets` field, you will need to provide credentials to
+the webhook using one of the following mechanisms:
+1. Create secret resources in any namespace of your preference that grants read
+access to the location of your container images and signatures and include it
+as part of your policy configuration.
+1. Create secret resources in and include them in the `image-policy-registry-credentials`
```

Comment:
I think the following sentence is incomplete: "Create secret resources in", maybe the "in" shouldn't be there or there something else that should be added to finish the sentence.

Reply:
Yes! Totally missed it.